File: algorithms/sorting/bubble-sort/bubble_sort.go
package main

import "fmt"

var arr = []int{8,2,4,3}

func bubble_sort(arr []int)
{
	for i := 0; i < len(arr); i++
	{
		for j := 0; j < len(arr) - 1 - i; j++
		{
			if arr[j] > arr[j+1]
			{
				arr[j], arr[j+1] = arr[j+1], arr[j]
			}
		}
	}
}

func main()
{
	bubble_sort(arr)
}

File: content/algorithms/sorting/bubble-sort/code/bubble-sort.c
#include <stdio.h>

void bubbleSort(int arr[]) 
{
    for (int i = 0; i < sizeof(arr) / sizeof(arr[0]) - 1; i++) 
    {
        for (int j = 0; j < sizeof(arr) / sizeof(arr[0]) - i - 1; j++) 
        {
            if (arr[j] > arr[j + 1]) 
            {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

void printArray(int arr[]) 
{
    for (int i = 0; i < sizeof(arr) / sizeof(arr[0]); i++) 
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main() 
{
    int arr[] = {64, 34, 25, 12, 22, 11, 90};
    printf("Unsorted array: ");
    printArray(arr);

    bubbleSort(arr);

    printf("Sorted array: ");
    printArray(arr);
    
    return 0;
}
